json-schemas/interface-methods/records-filter.json

```diff
@@ -47,6 +53,51 @@
           "$ref": "https://identity.foundation/dwn/json-schemas/defs.json#/definitions/date-time"
         }
       }
+    },
+    "datePublished": {
+      "type": "object",
+      "minProperties": 1,
+      "additionalProperties": false,
+      "properties": {
+        "from": {
+          "$ref": "https://identity.foundation/dwn/json-schemas/defs.json#/definitions/date-time"
+        },
+        "to": {
+          "$ref": "https://identity.foundation/dwn/json-schemas/defs.json#/definitions/date-time"
+        }
+      }
+    },
+    "dateUpdated": {
+      "type": "object",
+      "minProperties": 1,
+      "additionalProperties": false,
+      "properties": {
+        "from": {
+          "$ref": "https://identity.foundation/dwn/json-schemas/defs.json#/definitions/date-time"
+        },
+        "to": {
+          "$ref": "https://identity.foundation/dwn/json-schemas/defs.json#/definitions/date-time"
+        }
+      }
+    }
+  },
+  "dependencies": {
+    "datePublished": {
```

Comment:
For my understanding: is this just a perf optimization or is there anything broken if we don't have this dependency check? Wouldn't the outcome just that there are zero entries returned?

Fair enough if this is just a silly query prevention mechanism, just want to make sure I understand the intent.

Reply:
If this check wasn't in place it would actually return only `published` records, because we automatically inject `published: true` when you are filtering by `datePublished`.

So since the nonsensical intent of the query isn't clear, I think it's better to reject it rather than return published results.

Reply:
Got it. Good that I asked, 'coz I thought it was an perf optimization thing.